### PR TITLE
Add Secure View for Input View

### PR DESCRIPTION
**Pull request by ehab-al-cko**
## Proposed changes

Use Secure View for Expiry Date

## Types of changes

What types of changes does your code introduce to Frames Ios?
_Put an `x` in the boxes that apply_

* [ ] Bugfix (non-breaking change which fixes an issue)
* [ ] New feature (non-breaking change which adds functionality)
* [ ] Breaking change (fix or feature that would cause existing functionality to not work as expected)

## Checklist

_Put an `x` in the boxes that apply. You can also fill these out after creating the PR. If you're unsure about any of them, don't hesitate to ask. We're here to help! This is simply a reminder of what we are going to look for before merging your code._

* [ ] Lint and unit tests pass locally with my changes
* [ ] I have added tests that prove my fix is effective or that my feature works
* [ ] I have added necessary documentation (if appropriate)

## Further comments

If this is a relatively large or complex change, kick off the discussion by explaining why you chose the solution you did and what alternatives you considered, etc...

### CODE REVIEW

File: Source/Extensions/UIViewExtensions.swift [hunk 1]
```diff
@@ -1,6 +1,14 @@
 import Foundation
 import UIKit
 
+@objc protocol Updatable {
+    func update(with style: Any?)
+}
+
+extension UIView: Updatable {
+    func update(with style: Any?) {}
```
Comment (alex-ioja-yang-cko) [inline Source/Extensions/UIViewExtensions.swift:9]: Why do we need this protocol? Why can't the concrete object just hold the method and we want it accessed via some protocol interface?

**Review by alex-ioja-yang-cko — APPROVED** [review]
Cool

**Comment by alex-ioja-yang-cko** [comment]
Safari Tabs 1 
Alex memory 0
🤕

File: Tests/UI/New UI/BillingForm/View/ExpiryDateViewTests.swift [hunk 2]
```diff
@@ -87,5 +89,4 @@ class ExpiryDateViewTests: XCTestCase {
         view.updateExpiryDate(to: "Test")
         XCTAssertFalse(view.style?.error?.isHidden ?? true)
     }
-
-}
+}
```
Comment (harry-brown-cko) [inline Tests/UI/New UI/BillingForm/View/ExpiryDateViewTests.swift:92]: not a blocker, but we should have newlines at the end of our files

**Review by harry-brown-cko — APPROVED** [review]
lgtm

File: Tests/UI/New UI/BillingForm/View/ExpiryDateViewTests.swift (hunk 2 above)
Comment (ehab-al-cko) [reply Tests/UI/New UI/BillingForm/View/ExpiryDateViewTests.swift:92]:
```suggestion
}

```

File: Source/UI/NewUI/CommonUI/View/Core/SecureDisplayView.swift [hunk 3]
```diff
@@ -0,0 +1,98 @@
...
+//  Created by Alex Ioja-Yang on 
+//  Copyright © 2022 Checkout. All rights reserved.
+//
+
+import UIKit
+
+/**
+Secure display view preventing leaking of content
+ */
+public final class SecureDisplayView: UIView {
```
Comment (deepesh-vasthimal-cko) [inline Source/UI/NewUI/CommonUI/View/Core/SecureDisplayView.swift:14]: Does this need to be public, I am hoping the components we develop are using SecureDisplayView internally our merchants will not be interested to use SecureDisplayView from external.

Comment (deepesh-vasthimal-cko) [reply Source/UI/NewUI/CommonUI/View/Core/SecureDisplayView.swift:14]: Also we need to add some unit tests here for SecureDisplayView and what is expected. we can refer the internal library for SecureDisplayView for same.

File: Source/UI/NewUI/CommonUI/View/Core/TextFieldView.swift [hunk 4]
```diff
@@ -31,23 +31,22 @@ class TextFieldView: UIView {
...
     @objc func textFieldEditingChanged(textField: UITextField) {
         delegate?.textFieldShouldChangeCharactersIn(textField: textField, replacementString: "")
     }
     
     private func setupConstraintsInOrder() {
-        guard let textField = textField else { return }
-        addSubview(textField)
-        textField.setupConstraintEqualTo(view: self)
+        let securedTextField = SecureDisplayView(secure: textField).disabledAutoresizingIntoConstraints()
+        addSubview(securedTextField)
```
Comment (deepesh-vasthimal-cko) [inline Source/UI/NewUI/CommonUI/View/Core/TextFieldView.swift:48]: 👍

**Review by deepesh-vasthimal-cko — COMMENTED** [review]
Few comments on SecureDisplayView unit tests

**Pull request merged**